src/ducks/context/BanksContext.jsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import { Q } from 'cozy-client'
+import { KONNECTOR_DOCTYPE } from 'doctypes'
+import { useJobsContext } from 'ducks/context/JobsContext'
+
+export const BanksContext = createContext({})
+
+export const useBanksContext = () => {
+  return useContext(BanksContext)
+}
+
+/**
+ * BanksProvider
+ *
+ * @param children
+ * @param client
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const BanksProvider = ({ children, client }) => {
+  const { jobsInProgress = [] } = useJobsContext()
+  const [banksJobsInProgress, setBanksJobsInProgress] = useState([])
+
+  useEffect(() => {
+    const arr = [...jobsInProgress]
+    if (arr.length > 0) {
+      const formatJobsInProgress = async () => {
+        const ids = arr.map(jobsInProgress => {
+          const slug = jobsInProgress.konnector
+          return `${KONNECTOR_DOCTYPE}/${slug}`
+        })
+        const resp = await client.query(Q(KONNECTOR_DOCTYPE).getByIds(ids))
```

Comment:
Nit: Seems like we will refetch all the running konnectors every time there is a change of running jobs. Not sure this is necessary, an improvement could be to check in the client state if we already have the ids.

```
const idsToFetch = ids.filter(id => !client.getDocumentFromState(KONNECTOR_DOCTYPE, id))
```

Reply:
Indeed, it can be optimized.